lib/junk_drawer/notifier.rb

```diff
@@ -29,7 +29,13 @@ def call(*args)
   private
 
     def strategy
-      STRATEGIES.fetch(self.class.strategy)
+      chosen_strategy = self.class.strategy
+
+      if chosen_strategy.is_a?(Class) && chosen_strategy.respond_to?(:call)
```

Comment:
I think we can do without the first part of the check here. It could be any old callable thing, like a `lambda`.

Reply:
Ah yep good point. I initially had it also explicitly checking if it was a `Proc` (with a case statement), but I nixed that part. Didn't quite get to the next logical step of "anything with a call method" but that's what I was looking for.